recipes/opentdf-client/all/conanfile.py

```diff
@@ -51,17 +51,25 @@ def export_sources(self):
             self.copy(data["patch_file"])
 
     def validate(self):
+        # check minimum cpp standard supported by compiler
         if self.settings.compiler.get_safe("cppstd"):
             check_min_cppstd(self, self._minimum_cpp_standard)
+        # check minimum version of compiler
         min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
         if not min_version:
-            self.output.warn(f"{self.name} recipe lacks information about the {self.settings.compiler} compiler support.")
+            self.output.warn(f'{self.name} recipe lacks information about the {self.settings.compiler} compiler support.')
         else:
             if Version(self.settings.compiler.version) < min_version:
-                raise ConanInvalidConfiguration(f"{self.name} requires C++{self._minimum_cpp_standard} support."
-                                                 "The current compiler {self.settings.compiler} {self.settings.compiler.version} does not support it.")
+                raise ConanInvalidConfiguration(f'{self.name} requires {self.settings.compiler} {self.settings.compiler.version} but found {min_version}')
+        # Disallow MT and MTd
         if is_msvc_static_runtime(self):
-            raise ConanInvalidConfiguration("this package can not be built with MT or MTd at this time")
+            raise ConanInvalidConfiguration(f'{self.name} can not be built with MT or MTd at this time')
+        # Display warning if libiconv sneaks in somewhere
+        try:
+            if self.dependencies["libiconv"] and not self.options.allow_libiconv:
+                self.output.warn(f"libiconv has been found in dependencies tree along the option {self.name}:allow_libiconv=False")
```

Comment:
We do not like to rely on warnings, so the only required change it to raise an exception https://github.com/conan-io/conan-center-index/blob/9c5c6665ee5a1800d1e4b05ff23456c1f88ccaf8/docs/how_to_add_packages.md#verifying-dependency-options

This would most likely generate the wrong binary which would be wrong

Reply:
Sadly, that's a change that was not in there originally, but was suggested by a reviewer a few days ago:

https://github.com/conan-io/conan-center-index/pull/12697#discussion_r962635542

I'm OK with removing 67-70 if that helps.

Reply:
I know, that's why we have a 2 reviewer rule 😉

It's impossible for someone to remember all the rules and conventions even when they are documentation.

It should really be an error, or else the generated binaries would be incorrect and the option has no meaning